Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift

```diff
@@ -443,17 +443,20 @@ extension DefaultDiagnosticConsoleFormatter {
         } else {
             endColumn = sourceLine.count + 1
         }
-        
+
+        let sourceLineUTF8 = sourceLine.utf8
+
         let columnRange = startColumn..<endColumn
-        let startIndex = sourceLine.index(sourceLine.startIndex, offsetBy: columnRange.lowerBound - 1)
-        let endIndex = sourceLine.index(startIndex, offsetBy: columnRange.count)
+        let startIndex = sourceLineUTF8.index(sourceLineUTF8.startIndex, offsetBy: columnRange.lowerBound - 1)
+        let endIndex = sourceLineUTF8.index(startIndex, offsetBy: columnRange.count)
```

Comment:
Is `.count` right here? Should it be `.upperBound` instead?
```suggestion
        let endIndex = sourceLineUTF8.index(startIndex, offsetBy: columnRange.upperBound)
```

Reply:
Using `columnRange.upperBound` would give a "String index out of bounds" error since we are trying to get the indexes for the substring between `columnRange.lowerBound` and `columnRange.upperBound`.  

Since` columnRange.upperBound` stars at `1`:
> The column of the location, starting with 1.
 
We could maybe use 
```swift
      let startIndex = sourceLineUTF8.index(sourceLineUTF8.startIndex, offsetBy: columnRange.lowerBound - 1)
      let endIndex = sourceLineUTF8.index(sourceLineUTF8.startIndex, offsetBy: columnRange.upperBound - 1)
```
but I personally find:

```swift
      let startIndex = sourceLineUTF8.index(sourceLineUTF8.startIndex, offsetBy: columnRange.lowerBound - 1)
      let endIndex = sourceLineUTF8.index(startIndex, offsetBy: columnRange.count)
```
more concise, I'm happy to change it though. What do you think?

Reply:
I missed that the second index was relative to the first index—and not relative to the start. 

No need to change it.